### PR TITLE
Fix: Remove Excessive Usage of `panic!` in memfs Metadata Files

### PR DESCRIPTION
**Pull request by Reto911**
This PR aims to fix a problem of excessive usage of `panic!` in `async_fuse/memfs/s3_metadata.rs`, raised by #376 .

### CODE REVIEW

**Comment by Reto911** [comment]
The usage of `panic!` calls has been reduced from 52 to 1, and the remain one represents an unimplemented function.

https://github.com/Reto911/datenlord/blob/e552dae60febf4a9f9e150363dfdffe58d8dff6e/src/async_fuse/memfs/s3_metadata.rs#L1330

**Comment by Reto911** [comment]
Besides, as some caller that returns `nix::Result` but not `DatenLordResult`, the returned `DatenLordResult` cannot be passed up. Therefore, those caller that returns `nix::Result` still panics when get an `Error`. There are 4 such `panic!` calls.

As `nix::Result` can be converted into `DatenLordResult`, this might be fixed in the future,

**Comment by xiaguan** [comment]
The content of the PR is much appreciated. However, please ensure that the commit titles adhere to the Conventional Commits specification, which you can find here: [Conventional Commits 1.0.0](https://www.conventionalcommits.org/en/v1.0.0/). The title of the PR should follow the same standard. We could consider using a CI tool to automatically enforce this.

**Review by xiaguan — DISMISSED** [review]

**Comment by Reto911** [comment]
> The content of the PR is much appreciated. However, please ensure that the commit titles adhere to the Conventional Commits specification, which you can find here: [Conventional Commits 1.0.0](https://www.conventionalcommits.org/en/v1.0.0/). The title of the PR should follow the same standard. We could consider using a CI tool to automatically enforce this.

Thank you for your reminding! I will pay attention on it in the future.

**Comment by xiaguan** [comment]
You can edit the commit messages and then force-push. It's alright.

File: src/async_fuse/memfs/s3_metadata.rs [hunk 1]
```diff
@@ -68,24 +72,29 @@ pub struct S3MetaData<S: S3BackEnd + Send + Sync + 'static> {
 }
 
 /// Parse S3 info
-fn parse_s3_info(info: &str) -> (&str, &str, &str, &str) {
+fn parse_s3_info(info: &str) -> DatenLordResult<(&str, &str, &str, &str)> {
     info.split(S3_INFO_DELIMITER)
         .next_tuple()
-        .unwrap_or_else(|| panic!("parse s3 information failed. s3_info: {info}"))
+        .ok_or_else(|| anyhow::anyhow!("parse s3 information failed. s3_info: {info}").into())
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_metadata.rs:78]: It's a very general input parsing error, we can add an error type for it, or maybe we've already have one.

File: src/async_fuse/memfs/s3_metadata.rs [hunk 2]
```diff
@@ -68,24 +72,29 @@ pub struct S3MetaData<S: S3BackEnd + Send + Sync + 'static> {
...
+        ino: u64,
+        fh: u64,
+        _flags: u32,
+        _lock_owner: u64,
+        flush: bool,
+    ) -> DatenLordResult<()> {
+        let mut inode = self
+            .get_node_from_kv_engine(ino)
+            .await?
+            .ok_or_else(|| error::build_inconsistent_fs(ino, "release"))?;
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_metadata.rs:97]: Maybe try https://stackoverflow.com/questions/38088067/equivalent-of-func-or-function-in-rust this way to get the function name.

File: src/async_fuse/memfs/s3_node.rs [hunk 3]
```diff
@@ -155,7 +155,11 @@ impl<S: S3BackEnd + Send + Sync + 'static> S3Node<S> {
                 for (name, serial_dir_entry) in dir {
                     let child_ino = serial_dir_entry.get_child_ino();
                     // Fetch the child node from kv
-                    let child_node = meta.get_node_from_kv_engine(child_ino).await;
+                    // TODO: Handle this Error
```
Comment (rogercloud) [inline src/async_fuse/memfs/s3_node.rs:158]: Please explain the reason to handle this later.

Comment (Reto911) [reply src/async_fuse/memfs/s3_node.rs]: I am sorry that this is my negligence, and it's fixed now.

File: src/async_fuse/memfs/s3_metadata.rs (hunk 2 above)
Comment (Reto911) [reply src/async_fuse/memfs/s3_metadata.rs]: Thank you for your suggestion. It's applied in my code now.

File: src/async_fuse/memfs/s3_metadata.rs (hunk 1 above)
Comment (Reto911) [reply src/async_fuse/memfs/s3_metadata.rs:84]: We do have one, called `S3Error`, but it's not designed well. We can refactor it in the future.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**